File: sandbox.client/front/ts/model/UserInfo.ts
import {PhoneType} from "./PhoneType";

export class UserInfo {
  public id: string;
  public accountName: string;
  public surname: string | null;
  public name: string | null;
  public patronymic: string | null;

  public charm: string | null;
  public age: number;
  public totalBalance: number;
  public minBalance: number;
  public maxBalance: number;

  public phoneType: PhoneType | null;

  public assign(o: any): UserInfo {
    this.id = o.id;
    this.accountName = o.accountName;
    this.surname = o.surname;
    this.name = o.name;
    this.patronymic = o.patronymic;
    this.phoneType = o.phoneType;

    this.charm = o.charm;
    this.age = o.age;
    this.totalBalance = o.totalBalance;
    this.minBalance = o.minBalance;
    this.maxBalance = o.maxBalance;

    return this;
  }

  public static copy(a: any): UserInfo {
    let ret = new UserInfo();
    ret.assign(a);
    return ret;
  }
}

File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/UserInfo.java
package kz.greetgo.sandbox.controller.model;

import kz.greetgo.sandbox.controller.model.PhoneType;

public class UserInfo {
  public String id;
  public String accountName;
  public String surname;
  public String name;
  public String patronymic;
  public PhoneType phoneType;
  public String charm;
  public int age;
  public int totalBalance;
  public int minBalance;
  public int maxBalance;

  public UserInfo(String id,
                  String surname,
                  String name,
                  String patronymic,
                  String charm,
                  int age,
                  int totalBalance,
                  int minBalance,
                  int maxBalance) {
    this.id = id;
    this.surname = surname;
    this.name = name;
    this.patronymic = patronymic;
    this.charm = charm;
    this.age = age;
    this.totalBalance = totalBalance;
    this.minBalance = minBalance;
    this.maxBalance = maxBalance;
  }

  public UserInfo() {

  }
}
